### PR TITLE
Toolbar actions improvements

### PR DESCRIPTION
**Pull request by LeoMash**
- Connection status icon moved to main toolbar.
- Added connection information - executable name, project name, process id
- Previous status widget in status bar removed
- Added some usability actions
- Attach debugger to connected editor
- Goto settings pages
- Show/Hide play actions on toolbar

### CODE REVIEW

File: src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt [hunk 1]
```diff
@@ -1,39 +1,204 @@
...
+            if (debugProcess is CidrDebugProcess) {
+                // TODO: need to retrieve target pid from process
+                val pid = 0 // debugProcess.processHandler
+                if (pid == connectionInfo.processId) {
+                    e.presentation.isEnabled = false
+                    return
+                }
+            }
+        }
+        */
```
Comment (DecoyRS) [inline src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt:111]: Is it here for the future functionality? I generally don't like dead commented code.

File: src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt [hunk 2]
```diff
@@ -174,6 +351,33 @@ class PlayMode : DumbAwareToggleAction() {
...
+    override fun getActionUpdateThread(): ActionUpdateThread = ActionUpdateThread.BGT
+
+    override fun update(e: AnActionEvent) {
+        super.update(e)
+
+        val host = e.getUnrealHost()
+        if (host == null) {
+            e.presentation.isEnabledAndVisible = false
+        }
+        e.presentation.isEnabledAndVisible = true
```
Comment (DecoyRS) [inline src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt:369]: Looks like there's a `return` missing.
But I'd prefer here (and similar places) `isEnabledAndVisible = host != null`

File: src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt (hunk 1 above)
Comment (DecoyRS) [reply src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt:111]: And we have the pid of the process now, mb we can uncomment it?

File: src/cpp/RiderLink/Source/RiderLink/Private/RiderLink.cpp [hunk 3]
```diff
@@ -13,17 +14,27 @@ DEFINE_LOG_CATEGORY(FLogRiderLinkModule);
...
+	ProtocolFactory.Reset();
 	UE_LOG(FLogRiderLinkModule, Verbose, TEXT("RiderLink SHUTDOWN FINISH"));
 }
 
 void FRiderLinkModule::StartupModule()
 {
 	UE_LOG(FLogRiderLinkModule, Verbose, TEXT("RiderLink STARTUP START"));
-	ProtocolFactory::InitRdLogging();
+	ProtocolFactory = MakeUnique<class ProtocolFactory>(GetProjectName());
+	ProtocolFactory->InitRdLogging();
```
Comment (DecoyRS) [inline src/cpp/RiderLink/Source/RiderLink/Private/RiderLink.cpp:37]: If it's a class now and this method is not used anywhere, mb it's better to be moved to the constructor?

**Review by DecoyRS — CHANGES_REQUESTED** [review]

File: src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt (hunk 1 above)
Comment (LeoMash) [reply src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt:111]: Yes, it is the code for the future functionality. We need to enumerate all current debugging sessions and find if we already attached debugger to the same pid as connected by unreal link - we want to disable this action as subsequent attaching will always fail with non-obvious error message (WinAPI things). 
And now there is no way to retrieve target pid from CidrDebugProcess to compare with, and this improvement is not that I want to push to 223 (as ff already happened), so I will implement it in 231. But I don't want to drop code for abovementioned logic, so I commented so someone else can "finish the job" when I'll be not available anymore.

**Review by DecoyRS — APPROVED** [review]

**Pull request merged**